src/librustc_codegen_ssa/back/linker.rs

```diff
@@ -416,6 +416,7 @@ impl<'a> Linker for GccLinker<'a> {
                 if let Some(implib_name) = implib_name {
                     let implib = out_filename.parent().map(|dir| dir.join(&implib_name));
                     if let Some(implib) = implib {
+                        self.linker_arg("--kill-at");
```

Comment:
Ok, I mostly understand what this option does from the doc, but still not sure why we are using it.

Reply:
Ah, I see the explanation in the top comment.
It's still not clear why `ld` without `--kill-at` is compatible with `lld` with `--kill-at`.
Perhaps MinGW builds of `ld` enable this flag by default or something?

cc https://reviews.llvm.org/D44292

Reply:
LD seems to use heuristic to enable it on it's own:
```
D:/msys64/mingw32/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld.exe: warning: resolving _SomeFunction by linking to _SomeFunction@0
    Use --enable-stdcall-fixup to disable these warnings
    Use --disable-stdcall-fixup to disable these fixups
```
LLD doesn't even have `--{enable,disable}-stdcall-fixup`.

Reply:
I can drop this change until we know better what is going on.

Reply:
Leaving this at your discretion.
(Binutils code is available, so it can be figured out how the heuristic works, but I probably won't read it myself in the near future.)

Reply:
LD isn't very smart in that regard: https://github.com/bminor/binutils-gdb/blob/f4a220077b03af3a1f905b7dc6dc84c0a06d582f/ld/emultempl/pe.em#L1077-L1122

Turns out my previous comments about https://github.com/rust-lang/rust/issues/50176 were sort of correct. LD and LLD can successfully link it when import lib is created by LLD **without** `--kill-at`.
Dropped this commit since LLD seems more correct fixing this mess is out of scope of this PR.